src/main/scala/path/Path.scala

```diff
@@ -90,6 +98,25 @@ case class Path(start: NodeType, pieces: Seq[PathPiece] = Seq.empty) extends Ref
 
     pathIdentifier + start.toQuery(referenceableMap) + (pieces map (_.toQuery(referenceableMap)) mkString "")
   }
+
+  private[scalypher] def replaceNode(oldNode: Node, newNode: Node): Path = {
```

Comment:
these are part of some funky logic with assigning identifiers when creating relationships that i'll point out